### PR TITLE
Add Season 8 pubkeys and update hardfork constants

### PR DESCRIPTION
**Pull request by smk762**
Harvested from https://docs.google.com/spreadsheets/d/1qcRn6YF_UdjeSJxSz80nS1cqpqS-5x0NnGKo9VJPPq8/edit?usp=sharing

Related dPoW PR at https://github.com/KomodoPlatform/dPoW/pull/563

### CODE REVIEW

**Comment by DeckerSU** [comment]
These changes should go directly into the `patch-s8-prepare` branch, and then it should be merged with `dev`. The `patch-s8-prepare` branch already has everything needed, including constants, etc. The main purpose of this branch is to have a state where we only need to replace the pubkeys.

**Review by naezith — APPROVED** [review]

**Review by onur-ozkan — APPROVED** [review]

**Review by cipig — APPROVED** [review]

**Review by dimxy — APPROVED** [review]

**Review by shamardy — APPROVED** [review]

**Review by gcharang — APPROVED** [review]

**Comment by DeckerSU** [comment]
Conflicts have been resolved [here](https://github.com/KomodoPlatform/komodo/pull/620/commits/750d79c291b5dc0e769f96fafb9f2a136b0c2a68). To finalize the changes in `dev`, we need to approve the following:

1. PR #613 (`patch-remove-libsnark` -> `dev`)
2. PR #618 (`patch-macos-build-clang` -> `patch-remove-libsnark`)
3. PR #619 (`patch-data-carrier` -> `dev`)
4. PR #620 (`add/s8-pubkeys` -> `patch-s8-prepare`, which already contains `patch-remove-libsnark`, `patch-macos-build-clang`, and KIP0002 + KIP0003 changes)

Afterward, we need to create a final PR to merge the remaining changes into `dev`, with final checks. I can handle the merge and resolve any potential conflicts myself, but I'll need the reviewers' approval of PRs (1)-(4) first.

**Review by ca333 — APPROVED** [review]

File: src/komodo_hardfork.cpp [hunk 1]
```diff
@@ -17,8 +17,8 @@ const int32_t nS6HardforkHeight = 2963330;  // dPoW Season 6, Fri Jun 24 2022
 const uint32_t nS7Timestamp = 1688132253;   // dPoW Season 7, Fri Jun 30 2023 13:37:33 GMT+0000
 const int32_t nS7HardforkHeight = 3484958;  // dPoW Season 7, Fri Jun 30 2023
 
-const uint32_t nS8Timestamp = 1726321053;   // dPoW Season 8, Sat Sep 14 2024 13:37:33 GMT+0000
-const int32_t nS8HardforkHeight = 4100333;  // dPoW Season 8, Sat Sep 14 2024
+const uint32_t nS8Timestamp = 1728049053;   // dPoW Season 8, Fri Oct 4 2024 13:37:33 GMT+0000
+const int32_t nS8HardforkHeight = 4125988;  // dPoW Season 8, Fri Oct 4 2024 (est. via https://github.com/KomodoPlatform/NotaryNodes/blob/master/utils/estimate_hf_block.py)
```
Comment (ca333) [inline src/komodo_hardfork.cpp:21]: current estimation for height `4125988` is October 1st

cc @DeckerSU @smk762

Comment (smk762) [reply src/komodo_hardfork.cpp:21]: Ideally this should be (re)calculated as late as possible to make it as accurate as possible wrt to recent/anticipated block production rate.

Comment (ca333) [reply src/komodo_hardfork.cpp:21]: ok, will recalc tonight and include in https://github.com/KomodoPlatform/komodo/pull/623

Comment (DeckerSU) [reply src/komodo_hardfork.cpp:21]: I created my own script for these calculations, which you can find here: [estimate-blocks.py](https://github.com/DeckerSU/komodo_scripts/blob/master/estimate-blocks.py). According to my calculations, we have the following:
```
- Average blocks per day: 1370.96
- Minimum blocks per day: 1260
- Maximum blocks per day: 1457
- Current height: 4063104
- Days to future: 45
- Estimated block height at 2024-10-04 13:37:33: 4124797
```
Based on this, `October 4th` corresponds to approximately block `4124797` on average. Currently, we have `4125988` hardcoded, which should occur later than October 4th. So, my guess is that `4125988` is okay.

Comment (ca333) [reply src/komodo_hardfork.cpp:21]: thanks for validating @DeckerSU - lets keep the height as is then

**Pull request merged**